core/src/test/scala/org/http4s/parser/AcceptHeaderSpec.scala

```diff
@@ -20,61 +19,57 @@ class AcceptHeaderSpec extends Specification with HeaderParserHelper[Accept] wit
 
     "Parse all registered MediaRanges" in {
       // Parse a single one
-      parse("image/*").values.head must be_==(`image/*`)
+      parse("image/*").values.head must be_===(MediaRangeAndQValue(`image/*`))
 
       // Parse the rest
       foreach(MediaRange.snapshot.values) { m =>
         val r = parse(m.renderString).values.head
-        r must be_==(m)           // Structural equality
-        (r eq m) must be_==(true) // Reference equality
+        r must be_===(MediaRangeAndQValue(m))
       }
     }
 
     "Deal with '.' and '+' chars" in {
       val value = "application/soap+xml, application/vnd.ms-fontobject"
-      val accept = Accept(`application/soap+xml`, `application/vnd.ms-fontobject`)
-      parse(value) must be_==(accept)
+      val accept = Accept(MediaRangeAndQValue(`application/soap+xml`), MediaRangeAndQValue(`application/vnd.ms-fontobject`))
+      parse(value) must be_===(accept)
 
     }
 
     "Parse all registered MediaTypes" in {
       // Parse a single one
-      parse("image/jpeg").values.head must be_==(`image/jpeg`)
+      parse("image/jpeg").values.head must be_===(MediaRangeAndQValue(`image/jpeg`))
 
       // Parse the rest
       foreach(MediaType.snapshot.values) { m =>
         val r = parse(m.renderString).values.head
-        r must be_==(m)           // Structural equality
-        (r eq m) must be_==(true) // Reference equality
+        r must be_===(MediaRangeAndQValue(m))
       }
     }
 
     "Parse multiple Ranges" in {
       // Just do a single type
-      val accept = Accept(`audio/*`, `video/*`)
-      parse(accept.value) must be_==(accept)
+      val accept = Accept(MediaRangeAndQValue(`audio/*`), MediaRangeAndQValue(`video/*`))
```

Comment:
Because I'm compiling this in my head, this change is an example of what becomes necessary without that implicit, right?